sim/warlock/demonology/hand_of_guldan.go

```diff
@@ -0,0 +1,74 @@
+package demonology
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/warlock"
+)
+
+func (demonology *DemonologyWarlock) CurseOfGuldanDebuffAura(target *core.Unit) *core.Aura {
+	return target.GetOrRegisterAura(core.Aura{
+		Label:    "CurseOfGuldan-" + demonology.Label,
+		ActionID: core.ActionID{SpellID: 86000},
+		Duration: time.Second * 15,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			//TODO: Implement Crit rating for pet vs this target only
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			//TODO: Implement Crit rating for pet vs this target only
+		},
+	})
+}
+
+// TODO: Since we are assuming hand of guldan will hit all targets can we just give the pets 10% crit for the duration?
+func (demonology *DemonologyWarlock) registerHandOfGuldanSpell() {
+	if !demonology.Talents.Haunt {
+		return
+	}
+
+	demonology.RegisterAura(core.Aura{
+		Label:    "Hand of Guldan",
+		ActionID: core.ActionID{SpellID: 47197},
+		Duration: time.Second * 10,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+		},
+	})
+
+	//TODO: Damage is shadow flame... How to make this shadow and fire schools?
+	demonology.HandOfGuldan = demonology.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 71521},
+		SpellSchool:    core.SpellSchoolFire,
```

Comment:
`SpellSchool:    core.SpellSchoolFire | core.SpellSchoolShadow` but we will have to port multischool support from the SoD repo as its currently not supported in here